### PR TITLE
ppsspp: fix build against ffmpeg 4.4

### PR DESCRIPTION
**Pull request by lunik1**
###### Motivation for this change

Applying the same fix for compilation against ffmpeg 4.4 that I used for the libretro core in #123842.

I can't verify the resulting binary works. It crashes with an OpenGL error for me, but so does ppsspp on 20.09 so it's something on my end.

edit to ping the ppsspp maintainer: @AndersonTorres

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/ppsspp/default.nix [hunk 1]
```diff
@@ -26,6 +27,14 @@ mkDerivation rec {
     sha256 = "sha256-vfp/vacIItlPP5dR7jzDT7oOUNFnjvvdR46yi79EJKU=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "fix_ffmpeg_4.4.patch"; # to be removed with next release
```
Comment (AndersonTorres) [inline pkgs/misc/emulators/ppsspp/default.nix:32]: The comment is to be put before the patch definition, namely, two lines above.

But I will not complain on this now.

**Comment by AndersonTorres** [comment]
I will try to test this tomorrow

File: pkgs/misc/emulators/ppsspp/default.nix [hunk 2]
```diff
@@ -26,6 +27,14 @@ mkDerivation rec {
...
+      url = "https://patch-diff.githubusercontent.com/raw/hrydgard/ppsspp/pull/14176.patch";
+      sha256 = "sha256-ecDoOydaLfL6+eFpahcO1TnRl866mZZVHlr6Qrib1mo=";
+    })
+  ];
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ppsspp/default.nix:36]:
```suggestion
     # fix compability with ffmpeg 4.4, remove on next release after 1.11
    (fetchpatch {
      name = "fix_ffmpeg_4.4.patch";
      url = "https://patch-diff.githubusercontent.com/raw/hrydgard/ppsspp/pull/14176.patch";
      sha256 = "sha256-ecDoOydaLfL6+eFpahcO1TnRl866mZZVHlr6Qrib1mo=";
    })
  ];
```

**Comment by AndersonTorres** [comment]
```20:19:483 Core/Config.cpp:1306 I[LOADER]: Loading controller config: /home/anderson/.config/ppsspp/PSP/SYSTEM/controls.ini
20:19:484 Core/Config.cpp:1339 I[LOADER]: Config loaded: '/home/anderson/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
QObject::connect: No such slot MainWindow::frameSkippinGroup_triggered(QAction *)
20:20:125 Qt/QtMain.cpp:572 I[SYSTEM]: Initializing GL graphics context
20:20:125 OpenGL/GLFeatures.cpp:187 W[G3D]: Unknown GL vendor: 'X.Org'
20:20:125 OpenGL/GLFeatures.cpp:194 I[G3D]: GPU Vendor : X.Org ; renderer: AMD RAVEN (DRM 3.35.0, 5.4.100, LLVM 9.0.1) version str: 4.6 (Core Profile) Mesa 20.1.10 ; GLSL version str: 4.60
20:20:125 Qt/QtMain.cpp:574 I[SYSTEM]: Using thread, starting emu thread
20:20:125 OpenGL/GLRenderManager.cpp:133 I[G3D]: Updating inflight frames to 3
20:20:125 Core/Core.cpp:194 I[G3D]: pixel_res: 200x80. Calling NativeResized()
20:20:125 UI/NativeApp.cpp:1386 I[G3D]: NativeResized - setting flag
20:20:125 UI/NativeApp.cpp:836 I[SYSTEM]: NativeInitGraphics
20:20:127 UI/NativeApp.cpp:905 I[SYSTEM]: NativeInitGraphics completed
20:20:127 UI/DiscordIntegration.cpp:59 I[SYSTEM]: Discord connection initialized
20:20:172 Core/Core.cpp:194 I[G3D]: pixel_res: 960x544. Calling NativeResized()
20:20:172 UI/NativeApp.cpp:1386 I[G3D]: NativeResized - setting flag
20:20:172 UI/NativeApp.cpp:399 E[LOADER]: Failed graphics backends: OPENGL,ALL,ALL,ALL
20:20:172 Core/Config.cpp:624 E[LOADER]: Graphics backend failed for 0, trying another
20:20:172 Core/Config.cpp:649 E[LOADER]: All graphics backends failed
20:20:182 Core/Config.cpp:1420 I[LOADER]: Config saved (got storage permission): '/home/anderson/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
20:20:183 Core/Config.cpp:1433 I[LOADER]: Controller config saved: /home/anderson/.config/ppsspp/PSP/SYSTEM/controls.ini
20:20:229 UI/NativeApp.cpp:1112 I[G3D]: Resized flag set - recalculating bounds
20:20:229 UI/Screen.cpp:132 I[SYSTEM]: ScreenManager::resized(dp: 1707x967)
amdgpu: os_same_file_description couldn't determine if two DRM fds reference the same file description.
If they do, bad things may happen!
X Error of failed request:  BadAccess (attempt to access private resource denied)
  Major opcode of failed request:  149 (GLX)
  Minor opcode of failed request:  5 (X_GLXMakeCurrent)
  Serial number of failed request:  98
  Current serial number in output stream:  98
20:20:583 UI/DiscordIntegration.cpp:43 E[SYSTEM]: Discord destructor running though g_Discord.Shutdown() has not been called.

```

**Comment by AndersonTorres** [comment]
Maybe because I am using Wayfire?

**Comment by lunik1** [comment]
Could be, I had a similar issue on sway. Does installing ppsspp from 20.09 work for you?

**Comment by AndersonTorres** [comment]
Rigorously the same. No difference.

```
50:05:820 Core/Config.cpp:1306 I[LOADER]: Loading controller config: /home/anderson/.config/ppsspp/PSP/SYSTEM/controls.ini
50:05:821 Core/Config.cpp:1339 I[LOADER]: Config loaded: '/home/anderson/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
QObject::connect: No such slot MainWindow::frameSkippinGroup_triggered(QAction *)
50:06:056 Qt/QtMain.cpp:572 I[SYSTEM]: Initializing GL graphics context
50:06:056 OpenGL/GLFeatures.cpp:187 W[G3D]: Unknown GL vendor: 'X.Org'
50:06:056 OpenGL/GLFeatures.cpp:194 I[G3D]: GPU Vendor : X.Org ; renderer: AMD RAVEN (DRM 3.35.0, 5.4.100, LLVM 9.0.1) version str: 4.6 (Core Profile) Mesa 20.1.10 ; GLSL version str: 4.60
50:06:057 Qt/QtMain.cpp:574 I[SYSTEM]: Using thread, starting emu thread
50:06:057 OpenGL/GLRenderManager.cpp:133 I[G3D]: Updating inflight frames to 3
50:06:057 Core/Core.cpp:194 I[G3D]: pixel_res: 200x80. Calling NativeResized()
50:06:057 UI/NativeApp.cpp:1386 I[G3D]: NativeResized - setting flag
50:06:057 UI/NativeApp.cpp:836 I[SYSTEM]: NativeInitGraphics
50:06:059 UI/NativeApp.cpp:905 I[SYSTEM]: NativeInitGraphics completed
50:06:059 UI/DiscordIntegration.cpp:59 I[SYSTEM]: Discord connection initialized
50:06:063 Core/Core.cpp:194 I[G3D]: pixel_res: 960x544. Calling NativeResized()
50:06:063 UI/NativeApp.cpp:1386 I[G3D]: NativeResized - setting flag
50:06:064 UI/NativeApp.cpp:399 E[LOADER]: Failed graphics backends: OPENGL,ALL,ALL,ALL,ALL,ALL,ALL
50:06:064 Core/Config.cpp:624 E[LOADER]: Graphics backend failed for 0, trying another
50:06:064 Core/Config.cpp:649 E[LOADER]: All graphics backends failed
50:06:075 Core/Config.cpp:1420 I[LOADER]: Config saved (got storage permission): '/home/anderson/.config/ppsspp/PSP/SYSTEM/ppsspp.ini'
50:06:076 Core/Config.cpp:1433 I[LOADER]: Controller config saved: /home/anderson/.config/ppsspp/PSP/SYSTEM/controls.ini
amdgpu: os_same_file_description couldn't determine if two DRM fds reference the same file description.
If they do, bad things may happen!
X Error of failed request:  BadAccess (attempt to access private resource denied)
  Major opcode of failed request:  149 (GLX)
  Minor opcode of failed request:  5 (X_GLXMakeCurrent)
  Serial number of failed request:  98
  Current serial number in output stream:  98
50:06:125 UI/DiscordIntegration.cpp:43 E[SYSTEM]: Discord destructor running though g_Discord.Shutdown() has not been called.
```

**Comment by AndersonTorres** [comment]
And the same occurs when I ran without patches/updates the stable version.

**Comment by lunik1** [comment]
Well at least this patch hasn't _necessarily_ broken it 🙃. Getting late for me but I'll see if I can get ppsspp to run over the weekend.

**Comment by lunik1** [comment]
No luck so far on Wayland. I won't have much time to look at this this week, but I think the next step is to try on an X server. If that works we can verify this patch and merge this, and open a new issue for Wayland?

**Comment by AndersonTorres** [comment]
YEP!

**Comment by lunik1** [comment]
Both ppsspp from 20.09 and this PR work for me when forcing XWayland with `env -u WAYLAND_DISPLAY`.

**Comment by Artturin** [comment]
`-DUSE_WAYLAND_WSI=ON` is needed for wayland to work but https://github.com/hrydgard/ppsspp/issues/13845

**Comment by AndersonTorres** [comment]
@lunik1 can you address the backport issue?

**Comment by lunik1** [comment]
I can, should this be squashed for the backport?

**Comment by AndersonTorres** [comment]
Usually backports are made via cherry-pick.

**Pull request merged**